### ISSUE TITLE
Table component not working with dynamic data

### ISSUE THREAD

**Issue by ecerroni** [ElemeFE/element#3124]
### OS/Browers version

Version 56.0.2924.87 (64-bit) | Linux

### Vue version

2.1.10

### Steps to reproduce
I'm using the table element with success when I pass an array with data `:tableData="staticArray"`
However when I use vue-apollo I got an error

`Cannot define property:$extra, object is not extensible.`

I'm not sure if it depends on the fact that the data array changes after retrieving from graphQL so that the table element throw this error because the array mutates

### What is Expected?
I'd expect the table to update accordingly

### What is actually happening?
Error: `Cannot define property:$extra, object is not extensible.`

**Comment by ankitg1602** [comment]
Hi @ecerroni , were you able to find the problem? Can you please post the solution if you did. Thanks.

**Comment by ecerroni** [comment]
@ankitg1602
I think it was because data by either Apollo (vue-apollo package ) or Vue 2.x (when passing down props to child components) or both is freeze.

Possible workarounds, whatever works best in specific scenarios
- Deep copy data you get from Apollo (Object.assign will not be enough for nested objects/arrays)
- store Apollo data in localStorage and work with that.

I'm facing that only when I prefill a form with data I got from the apollo client.

In conclusion it's not an issue related to this repo. 

That's why I closed it as soon as I got that.
